autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -0,0 +1,533 @@
+                                                 sample_weight=sample_weight)
+
+    def predict(self, X: Dict[str, Any],
+                batch_size: int = 1000) -> np.array:
+        new_X = np.ones((X['X_train']['val_indices'].shape[0], 1))
+        return super(MyDummyRegressor, self).predict(new_X).astype(np.float32)
+
+    def estimator_supports_iterative_fit(self) -> bool:  # pylint: disable=R0201
+        return False
+
+    def get_additional_run_info(self) -> None:  # pylint: disable=R0201
+        return None
+
+
+def fit_and_suppress_warnings(logger: PicklableClientLogger, model: BaseEstimator,
+                              X: Dict[str, Any], y: Any
+                              ) -> BaseEstimator:
+    def send_warnings_to_log(message, category, filename, lineno,
+                             file=None, line=None) -> None:
+        logger.debug('%s:%s: %s:%s',
+                     filename, lineno, category.__name__, message)
+        return
+
+    with warnings.catch_warnings():
+        warnings.showwarning = send_warnings_to_log
+        model.fit(X, y)
+
+    return model
+
+
+# TODO: adjust for differences in image and tabular data, also maybe time series but its low priority
+class AbstractEvaluator(object):
+    def __init__(self, backend: Backend, queue: Queue, metric: List[autoPyTorchMetric],
+                 configuration: Optional[Configuration] = None,
+                 seed: int = 1,
+                 output_y_hat_optimization: bool = True,
+                 num_run: Optional[int] = None,
+                 include: Optional[Dict[str, Any]] = None,
+                 exclude: Optional[Dict[str, Any]] = None,
+                 disable_file_output: bool = False,
+                 init_params: Optional[Dict[str, Any]] = None,
+                 budget: Optional[float] = None,
+                 budget_type: Optional[str] = None,
+                 logger_port: Optional[int] = None) -> None:
+
+        self.starttime = time.time()
+
+        self.configuration = configuration
+        # TODO: adjust backend for autopytorch backend
+        self.backend = backend
+        self.queue = queue
+
+        # TODO: use autopytorch datasets once PR is passed
+        self.datamanager: BaseDataset = self.backend.load_datamanager()
+        self.include = include
+        self.exclude = exclude
+
+        self.X_train, self.y_train = self.datamanager.train_tensors
+
+        if self.datamanager.test_tensors is not None:
+            self.X_test, self.y_test = self.datamanager.test_tensors
+        else:
+            self.X_test, self.y_test = None, None
+
+        self.metric = metric
+        self.task_type = STRING_TO_TASK_TYPES[self.datamanager.task_type]
+        self.output_type = STRING_TO_OUTPUT_TYPES[self.datamanager.output_type]
+        self.issparse = self.datamanager.issparse
+
+        self.seed = seed
+
+        self.output_y_hat_optimization = output_y_hat_optimization
```

Comment:
what is this variable for?

Reply:
actually I am also a bit unclear about what exactly it does, however, in the code, it saves Y_optimization to targets for the ensemble, see [here](https://github.com/ravinkohli/Auto-PyTorch_refactor/blob/d6bc09b78ebf99e99278d91efdda4ca07c306a39/autoPyTorch/evaluation/abstract_evaluator.py#L444)

Reply:
I see, it is just a flag wether to save the labels/targets, let's write that somewhere :)